geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/QueryFilterSplitterTest.scala

```diff
@@ -548,5 +545,19 @@ class QueryFilterSplitterTest extends Specification {
       compareOr(options.head.strategies.head.primary, bbox1, bbox2)
       options.head.strategies.head.secondary must beSome // note: filter is too complex to compare explicitly...
     }
+
+    "handle ORs between indexed attribute fields" >> {
+      val spec =
+        "attr1:Long:cardinality=high:index=true,attr2:Long:cardinality=high:index=true," +
+            "name:String:cardinality=high:index=true,dtg:Date,*geom:Point:srid=4326"
+      val sft = SimpleFeatureTypes.createType("multiAttr", spec)
+      sft.setIndices(GeoMesaFeatureIndexFactory.indices(sft))
+      val splitter = new FilterSplitter(sft, GeoMesaFeatureIndexFactory.create(null, sft, sft.getIndices))
+      val filter = f("(attr1=1 OR attr2=2 OR name='3') and dtg during 2020-01-01T00:00:00.000Z/2020-01-02T00:00:00.000Z")
+      val options = splitter.getQueryOptions(filter)
+      options must haveLength(2)
+      options.map(_.strategies.map(_.index.name)) must
+          containTheSameElementsAs(Seq(Seq(Z3Index.name), Seq.fill(3)(AttributeIndex.name)))
+    }
```

Comment:
Is the new system property needed for this to work?

Or do the indexed ORs get handled separate from the complex-not-completely-indexed-OR case?

Reply:
the system prop is disabled for all these tests

Reply:
Ahhhh...  my union card says that I have to ask for unit tests for the new system property....  How big of a PITA would be that be?

(I'm also interested in trying think through when the setting would be used.  It seems like we may not need it.)

Reply:
yeah, i could add one.

Reply:
added